rtgui/filmnegative.cc

```diff
@@ -263,31 +290,58 @@ bool FilmNegative::button1Pressed(int modifierKey)
     EditSubscriber::action = EditSubscriber::Action::NONE;
 
     if (listener) {
-        refSpotCoords.push_back(provider->posImage);
+        if (spotbutton->get_active()) {
+
+            refSpotCoords.push_back(provider->posImage);
+
+            if (refSpotCoords.size() == 2) {
+                // User has selected 2 reference gray spots. Calculating new exponents
+                // from channel values and updating parameters.
+
+                std::array<float, 3> newExps;
+
+                if (fnp->getFilmNegativeExponents(refSpotCoords[0], refSpotCoords[1], newExps)) {
+                    disableListener();
+                    // Leaving green exponent unchanged, setting red and blue exponents based on
+                    // the ratios between newly calculated exponents.
+                    redRatio->setValue(newExps[0] / newExps[1]);
+                    blueRatio->setValue(newExps[2] / newExps[1]);
+                    enableListener();
+
+                    if (listener && getEnabled()) {
+                        listener->panelChanged(
+                            evFilmNegativeExponents,
+                            Glib::ustring::compose(
+                                "Ref=%1\nR=%2\nB=%3",
+                                greenExp->getValue(),
+                                redRatio->getValue(),
+                                blueRatio->getValue()
+                            )
+                        );
+                    }
+                }
+
+                switchOffEditMode();
+            }
+
+        } else if (filmBaseSpotButton->get_active()) {
 
-        if (refSpotCoords.size() == 2) {
-            // User has selected 2 reference gray spots. Calculating new exponents
-            // from channel values and updating parameters.
+            std::array<float, 3> newBaseLev = {};
```

Comment:
Here you initialize, above you don't. I've no problem with *not* initializing if the getter signature is `bool getSomething(T& something)`.